core/zoe/contracts/automaticRefund.js

```diff
@@ -11,18 +11,26 @@ import harden from '@agoric/harden';
  * @param {contractFacet} zoe - the contract facet of zoe
  */
 export const makeContract = harden((zoe, terms) => {
-  let count = 0;
-  const automaticRefund = harden({
-    makeOffer: async escrowReceipt => {
-      const { offerHandle } = await zoe.burnEscrowReceipt(escrowReceipt);
-      count += 1;
-      zoe.complete(harden([offerHandle]));
-      return `The offer was accepted`;
-    },
-    getOffersCount: () => count,
-  });
+  const initialInviteHandle = harden({});
+  let offersCount = 0;
+
+  const makeSeat = inviteHandle =>
+    harden({
+      makeOffer: () => {
+        offersCount += 1;
+        zoe.complete(harden([inviteHandle]), terms.assays);
+        return `The offer was accepted`;
+      },
+      makeInvite: () => {
+        const newInviteHandle = harden({});
+        const seat = makeSeat(newInviteHandle);
+        return zoe.makeInvite(seat, newInviteHandle);
```

Comment:
Does the made Invite contain the inviteHandle? If so, `zoe.makeInvite` should create it rather than taking it as a parameter.

I'm confused by the lack of a `seatDesc` or anything for that role, so that the holder of an allegedInvite can (hopefully typically) look at the contract source and know which call to makeInvite made the invite they are allegedly holding.

Reply:
I see. You need the seat to make the invite (correct) and you need the seat to know the inviteHandle (which I don't understand yet), so you need the inviteHandle to exist before you make the invite.

However, the seat doesn't actually need the handle until one of its methods are invoked.  Depending on  why the seat needs the handle, you might be able to solve this simply by initializing a const variable from what `makeInvite` returns, where that variable is in scope for the seat code.